### ISSUE TITLE
New Linter: MaxLinesPerRule

### ISSUE THREAD

**Issue by chris-dura** [causes/scss-lint#166]
I don't believe there is currently a linter that does the "other half" of `SelectorDepth`...
In my current project, we have max nesting level of 3, but we also have a standard of keeping each rule block around 50-ish lines; otherwise, the entire code block doesn't fit in editor windows, and it's very easy to lose the context of the specific rule you're scanning.

```scss
.hello {
   ...

   .foo {
      ...



      // ... and a couple dozen lines later

      .bar {
         // easy to forget that this is actually `.hello .foo .bar`
      }
   }
}
```

This basically came up in the context of the new `MergeableSelector` linter, which should probably have a configuration option to *not* recommend merging selectors if the merged rule would exceed a specified number of lines.

**Comment by sds** [comment]
The original difficulty with this linter was the issue that the Sass parser doesn't really give us the source range of a rule set. Thus enforcing based on raw line count would not be possible.

Considering the intent of this feature request as I perceive it, the goal seems to be flagging complexity and suggesting that a rule set be broken down into reusable components.

I've implemented what I believe accomplishes this goal with the `PropertyCount` linter in 8ddfb6affeaeec8b4e74a59dbf4923044a96d95b, which lints based on the number of properties (allowing you to specify whether you want to include properties in nested rule sets in the count). Let me know your thoughts.

**Comment by bartveneman** [comment]
> Considering the intent of this feature request as I perceive it, the goal seems to be flagging complexity and suggesting that a rule set be broken down into reusable components.

Exactly that, in my opinion. Thanks!
